types/aws-lambda/trigger/appsync-resolver.d.ts

```diff
@@ -0,0 +1,44 @@
+import { Handler } from '../handler';
+
+export type AppSyncResolverHandler<T, V> = Handler<AppSyncResolverEvent<T>, V | V[]>;
+
+/**
+ * See https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html
+ *
+ * @param T type of the arguments
+ */
+export interface AppSyncResolverEvent<T> {
+    arguments: T;
+    identity?: AppSyncIdentityIAM | AppSyncIdentityCognito;
```

Comment:
Since you know at configuration time which of these it will be, it's useful to be able to declare which it will be so you can pull out specific values without casting - this is a small note though.